commons/src/main/java/fr/free/nrw/commons/category/CategorizationFragment.java

```diff
@@ -363,12 +330,73 @@ public void afterTextChanged(Editable editable) {
         return rootView;
     }
 
+    private void requestSearchResults() {
+
+        final CountDownLatch latch = new CountDownLatch(1);
+
+        Utils.executeAsyncTask(new PrefixUpdater(this) {
+            @Override
+            protected ArrayList<String> doInBackground(Void... voids) {
+                ArrayList<String> result = new ArrayList<String>();
+                try {
+                    result = super.doInBackground();
+                    latch.await();
+                }
+                catch (InterruptedException e) {
+                    Log.w(TAG, e);
+                    //Thread.currentThread().interrupt();
+                }
+                return result;
+            }
+
+            @Override
+            protected void onPostExecute(ArrayList<String> result) {
+                super.onPostExecute(result);
+
+                results.addAll(result);
+                Log.d(TAG, "Prefix result: " + result);
+                categoriesAdapter.notifyDataSetChanged();
```

Comment:
I don't think this will do anything if you just modify results without actually updating the adapter.